File: ios/ObjectiveC/Beacon/2-Notification/Starter/NotificationComplete/MyNotificationDelegate.m
//
//  MyNotificationDelegate.m
//  QuickComplete
//
//  Created by Connecthings on 09/11/2017.
//  Copyright © 2017 R&D Connecthings. All rights reserved.
//

#import "MyNotificationDelegate.h"

@implementation MyNotificationDelegate {
    AdtagBeaconManager *adtagBeaconManager;
}

-(id) initWithAdtagManager: (AdtagBeaconManager *) _adtagBeaconManager {
    self = [super init];
    if (self) {
        adtagBeaconManager = _adtagBeaconManager;
    }
    return self;
}

-(void) userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNotificationResponse:(UNNotificationResponse *)response withCompletionHandler:(void (^)(void))completionHandler {
    [adtagBeaconManager didReceivePlaceNotification:response.notification.request.content.userInfo];
    completionHandler();
}

@end

File: ios/SWIFT/Beacon/2-Notification/Complete/Complete/MyNotificationDelegate.swift
//
//  NotificationDelegate.swift
//  QuickStart
//
//  Created by Connecthings on 09/11/2017.
//  Copyright © 2017 R&D connecthings. All rights reserved.
//

import Foundation
import UserNotifications
import AdtagLocationBeacon

public class MyNotificationDelegate: NSObject, UNUserNotificationCenterDelegate {

    let adtagBeaconManager: AdtagBeaconManager

    public init(adtagBeaconManager: AdtagBeaconManager) {
        self.adtagBeaconManager = adtagBeaconManager
        super.init()
    }

    @available(iOS 10.0, *)
    public func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
        adtagBeaconManager.didReceivePlaceNotification(response.notification.request.content.userInfo)
    }

}
